src/gsy_e/models/strategy/external_strategies/load.py

```diff
@@ -291,58 +292,60 @@ def event_offer(self, *, market_id, offer) -> None:
         if self.should_use_default_strategy:
             super().event_offer(market_id=market_id, offer=offer)
 
-    def _bid_aggregator(self, arguments: Dict) -> Dict:
-        """Callback for the bid endpoint when sent by aggregator."""
-        response_message = ""
-        arguments, filtered_fields = self.filter_degrees_of_freedom_arguments(arguments)
-        if filtered_fields:
-            response_message = (
-                "The following arguments are not supported for this market and have been "
-                f"removed from your order: {filtered_fields}.")
-
-        required_args = {"price", "energy", "type", "transaction_id"}
-        allowed_args = required_args.union({"replace_existing",
-                                            "time_slot",
-                                            "attributes",
-                                            "requirements"})
-
+    def _offer_aggregator(self, arguments: Dict) -> Dict:
+        """Callback for the offer endpoint when sent by aggregator."""
         try:
             market = self._get_market_from_command_argument(arguments)
-            # Check that all required arguments have been provided
-            assert all(arg in arguments.keys() for arg in required_args)
-            # Check that every provided argument is allowed
-            assert all(arg in allowed_args for arg in arguments.keys())
-
-            replace_existing = arguments.pop("replace_existing", True)
-            assert self.can_bid_be_posted(
-                arguments["energy"],
-                arguments["price"],
-                self.state.get_energy_requirement_Wh(market.time_slot) / 1000.0,
-                market,
-                replace_existing=replace_existing)
-
-            bid = self.post_bid(
-                market,
-                arguments["price"],
-                arguments["energy"],
-                replace_existing=replace_existing,
-                attributes=arguments.get("attributes"),
-                requirements=arguments.get("requirements")
-            )
+            if self.area.is_market_settlement(market.id):
+                assert self.state.can_post_settlement_offer(market.time_slot), (
```

Comment:
I think that this assertion can fail even if the implementation is correct (because it depends on the user/client input). Since this is new code, I'd suggest using exceptions instead of assertions.

Reply:
Yeah, I was thinking the same, just copying from the old implementation. Will do like you propose.